lib/polymorphic_integer_type/activerecord_5_0_0/association_query_handler_extension.rb

```diff
@@ -0,0 +1,34 @@
+module PolymorphicIntegerType
+  module AssociationQueryHandlerExtension
+    def call(attribute, value)
+      queries = {}
+      table = value.associated_table
+
+      if value.base_class
+        queries[table.association_foreign_type.to_s] = polymorphic_value_for(value)
+      end
+
+      queries[table.association_foreign_key.to_s] = value.ids
+      predicate_builder.build_from_hash(queries)
+    end
+
+    protected
+
+    def polymorphic_value_for(value)
+      table = value.associated_table
+      association = table.send(:association)
+      klass = association.active_record
+      name = association.name
+
+      if klass.respond_to?("#{name}_type_mapping")
+        klass.send("#{name}_type_mapping").key(value.base_class.sti_name)
```

Comment:
I wonder if we need to do a similar lookup to what we did in the extension (i.e. looking up `sti_name`, `base_class.name`, etc)?

Reply:
yeah, I think that would be the safest way.

Reply:
👍 Fixed in 78f3e39.

I had a question about the ordering of what we look up. [Right now in the extension](https://github.com/doliveirakn/polymorphic_integer_type/blob/6949640ab42ea35efc1049e5844443db06b09df2/lib/polymorphic_integer_type/extensions.rb#L37-L39) we do:

1. `class.sti_name`
2. `class.base_class`
3. `class.base_class.sti_name`

Is there a reason we don't look it up in this order:

1. `class.sti_name`
2. `class.base_class.sti_name`
3. `class.base_class` (as a final fallback)